cyral/data_source_cyral_integration_logging.go

```diff
@@ -0,0 +1,90 @@
+package cyral
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cyralinc/terraform-provider-cyral/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type ListIntegrationLogsResponse struct {
+	Integrations []LoggingIntegration `json:"integrations"`
+}
+
+func (resp *ListIntegrationLogsResponse) WriteToSchema(d *schema.ResourceData) error {
+	var integrations []interface{}
+	for _, integration := range resp.Integrations {
+		// write in config scheme
+		configType, config, err := getLoggingConfig(&integration)
+		if err != nil {
+			return err
+		}
+
+		integrations = append(integrations, map[string]interface{}{
+			"id":                 integration.Id,
+			"name":               integration.Name,
+			"receive_audit_logs": integration.ReceiveAuditLogs,
+			configType:           config,
+		})
+	}
```

Comment:
You can avoid unnecessary allocations by allocating the slice ahead of time, since we know its length.

```suggestion
	integrations := make([]interface{}, len(resp.Integrations))
	for i, integration := range resp.Integrations {
		// write in config scheme
		configType, config, err := getLoggingConfig(&integration)
		if err != nil {
			return err
		}

		integrations[i] = map[string]interface{}{
			"id":                 integration.Id,
			"name":               integration.Name,
			"receive_audit_logs": integration.ReceiveAuditLogs,
			configType:           config,
		}
	}
```

Reply:
Used your proposed solution here: [35daafa](https://github.com/cyralinc/terraform-provider-cyral/pull/395/commits/35daafa7a278976ed6616a2964405590d94d3366). Thank you!